eth/backend.go

```diff
@@ -231,6 +232,15 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 		return nil, err
 	}
 
+	// Check for invalid shutdown
+	invalidShutdown, _ := chainDb.Get([]byte("unsafe-shutdown"))
+	if invalidShutdown != nil {
+		log.Error("unsafe shutdown detected", "time", string(invalidShutdown))
+	}
+	// Create an invalid shutdown in database in case the app crashed
+	if err = chainDb.Put([]byte("unsafe-shutdown"), []byte(time.Now().String())); err != nil {
+		log.Warn("Failed to record possible future unsafe shutdown", "err", err)
+	}
```

Comment:
The PR uses a single `unsafe-shutdown` key to store a single timestamp. That's a good start, but we need a bit more information, otherwise quirky scenarios get lost. Notably, it's important to retain past crashes too, so a user might see if there was a fault recently, not just in the last restart (user's tend to restart their nodes a few times until attempting to investigate an issue, so a restart cannot delete the cause).

A simplistic suggestion would be to use a slice of timestamps instead of a single timestamp as the value of the entry. When Geth starts, you can append the current time, and on shutdown, you can pop off the tail. That way on startup you can actually list all the recent crashes, not just the last one.